python/pyiceberg/catalog/glue.py

```diff
@@ -0,0 +1,247 @@
+            partition_spec: PartitionSpec for the table.
+            sort_order: SortOrder for the table.
+            properties: Table properties that can be a string based dictionary.
+        Returns:
+            Table: the created table instance
+        Raises:
+            AlreadyExistsError: If a table with the name already exists
+            ValueError: If the identifier is invalid
+        """
+        database_name, table_name = self.identifier_to_tuple(identifier)
+
+        location = self._resolve_table_location(location, database_name, table_name)
+        metadata_location = f"{location}/metadata/00000-{uuid.uuid4()}.metadata.json"
+        metadata = new_table_metadata(
+            location=location, schema=schema, partition_spec=partition_spec, sort_order=sort_order, properties=properties
+        )
+        io = load_file_io({**self.properties, **properties}, location=location)
+        _write_metadata(metadata, io, metadata_location)
+        try:
+            self.glue.create_table(
+                DatabaseName=database_name, TableInput=_construct_table_input(table_name, metadata_location, properties)
+            )
+        except self.glue.exceptions.AlreadyExistsException as e:
+            raise TableAlreadyExistsError(f"Table {database_name}.{table_name} already exists") from e
+        except self.glue.exceptions.EntityNotFoundException as e:
+            raise NoSuchNamespaceError(f"Database {database_name} not found") from e
+
+        try:
+            load_table_response = self.glue.get_table(DatabaseName=database_name, Name=table_name)
+        except self.glue.exceptions.EntityNotFoundException as e:
+            raise NoSuchTableError(f"Table {database_name}.{table_name} fail to be created") from e
+
+        glue_table = load_table_response[PROP_GLUE_TABLE]
+        return _convert_glue_to_iceberg(glue_table, io)
+
+    def load_table(self, identifier: Union[str, Identifier]) -> Table:
+        """Loads the table's metadata and returns the table instance.
+        You can also use this method to check for table existence using 'try catalog.table() except TableNotFoundError'
+        Note: This method doesn't scan data stored in the table.
+        Args:
+            identifier: Table identifier.
+        Returns:
+            Table: the table instance with its metadata
+        Raises:
+            NoSuchTableError: If a table with the name does not exist, or the identifier is invalid
+        """
+        database_name, table_name = self.identifier_to_tuple(identifier)
+        try:
+            load_table_response = self.glue.get_table(DatabaseName=database_name, Name=table_name)
+        except self.glue.exceptions.EntityNotFoundException as e:
+            raise NoSuchTableError(f"Table does not exists: {table_name}") from e
+        loaded_table = load_table_response[PROP_GLUE_TABLE]
+        io = load_file_io(self.properties, loaded_table[PROP_GLUE_TABLE_PARAMETERS][PROP_METADATA_LOCATION])
```

Comment:
IO instantiation looks fishy to me. We need to make sure we're doing it consistently. Might want to handle this in `_convert_glue_to_iceberg`.

Reply:
Thank you for your suggestion. I refactored the code to initialize the IO inside `_convert_glue_to_iceberg`. 

For consistency, I made the two io instantiations using `self.properties` and `metadata_location` to load the IO:
```
io = load_file_io(properties=self.properties, location=metadata_location)
```